src/System.Net.Requests/tests/RequestStreamTest.cs

```diff
@@ -11,11 +11,18 @@
 
 namespace System.Net.Tests
 {
-    [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework, "Tests hang on .NET Framework")]
     public class RequestStreamTest
     {
         readonly byte[] buffer = new byte[1];
 
+        static RequestStreamTest()
+        {
+            // On .NET Framework, the default limit for connections/server is very low (2).
+            // On .NET Core, the default limit is higher. Since these tests run in parallel,
+            // the limit needs to be increased to avoid timeouts when running the tests.
+            ServicePointManager.DefaultConnectionLimit = int.MaxValue;
```

Comment:
> because we never dispose the RequestStream

We do Dispose of the request stream: they're all (or at least the ones I looked at) wrapped in using blocks.  The problem is that there appears to be a bug in desktop where disposing of the request stream is insufficient to actually release the connection, and to do that you need to actually get the response so you can dispose of it.

I'm not sure how disposing it masks what we're trying to do with the request stream: we just create a helper and change a test like:
```C#
using (Stream stream = GetRequestStream())
{
    Assert.Throws<ArgumentNullException>("buffer", () => stream.Write(null, 0, 1));
}
```
to something like:
```C#
using (RequestStreamWrapper w = GetRequestStream())
{
    Assert.Throws<ArgumentNullException>("buffer", () => w.Stream.Write(null, 0, 1));
}
```
where the RequestStreamWrapper stores both the Stream and the HttpWebRequest, such that its Dispose (called implicitly by the using) can do GetResponse().Dispose on the request... it could even choose only to do this for desktop where there's the bug we're working around.